core/src/main/java/apoc/cypher/CypherInitializer.java

```diff
@@ -46,23 +52,31 @@ public void available() {
         Util.newDaemonThread(() -> {
 
             try {
-                awaitApocProceduresRegistered();
+                final boolean isSystemDatabase = db.databaseName().equals(GraphDatabaseSettings.SYSTEM_DATABASE_NAME);
+                if (!isSystemDatabase) {
+                    awaitApocProceduresRegistered();
+                }
                 Configuration config = dependencyResolver.resolveDependency(ApocConfig.class).getConfig();
 
-                TreeMap<String, String> initializers = Iterators.stream(config.getKeys(ApocConfig.APOC_CONFIG_INITIALIZER_CYPHER))
-                        .collect(Collectors.toMap(k -> k, k -> config.getString(k),
-                                (v1, v2) -> {
-                                    throw new RuntimeException(String.format("Duplicate key for values %s and %s", v1, v2));
-                                },
-                                TreeMap::new));
+                TreeMap<String, String> initializers = new TreeMap<>();
+
+                config.getKeys(ApocConfig.APOC_CONFIG_INITIALIZER + "." + db.databaseName()).forEachRemaining(key -> initializers.put(key, config.getString(key)));
+
+                if (!isSystemDatabase) {
+                    config.getKeys(ApocConfig.APOC_CONFIG_INITIALIZER_CYPHER).forEachRemaining(key -> initializers.put(key, config.getString(key)));
+                }
 
                 for (Object initializer : initializers.values()) {
                     String query = initializer.toString();
-                    try {
-                        db.executeTransactionally(query);
-                        userLog.info("successfully initialized: " + query);
-                    } catch (Exception e) {
-                        userLog.warn("error upon initialization, running: " + query, e);
+                    if (!query.isEmpty()) {
+                        try {
+                            // we need to apply a retry strategy here since in systemdb we potentially conflict with
+                            // creating contraints which could cause our query to fail with a transient error.
```

Comment:
what is an example of this?

Reply:
when not having the retry strategy I found that an initializer of `apoc.initializer.system=create user dummy set password 'abc'` resulted in a transaction failed since a concurrent transaction was setting up a constraint.